### PR TITLE
Migrate bicep example: 101/data-factory-v2-blob-to-blob-copy

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/101/data-factory-v2-blob-to-blob-copy

https://github.com/Azure/bicep/pull/3149

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@djpmsft - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by StephenWeatherford** [comment]
FYI @JFolberth

**Comment by azure-quickstarts** [comment]
@djpmsft - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by djpmsft** [comment]
I am not on the ADF team anymore so I don't think I am the right contact here

**Comment by StephenWeatherford** [comment]
@djpmsft was the sample referenced by documentation?  If not, perhaps we can just remove the ue contact.

File: quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep [hunk 1]
```diff
@@ -0,0 +1,142 @@
+@description('Data Factory Name')
+param dataFactoryName string = 'datafactory${uniqueString(resourceGroup().id)}'
+
+@description('Location of the data factory. Currently, only East US, East US 2, and West Europe are supported.')
+param location string = resourceGroup().location
```
Comment (bmoore-msft) [inline quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep:5]: if locations are restricted, remove the default and add allowedValues - though looking at the latest list I don't think that restriction is accurate.

File: quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    typeProperties: {
+      location: {
+        type: 'AzureBlobStorageLocation'
+        container: blobContainer.name
+        folderPath: 'input'
+        fileName: 'emp.txt'
+      }
+    }
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep:68]: Do we need this? Seems like we either don't need it or we need one for the blobContainer too...

Also the resource below has the same pattern and has no dependsOn property

File: quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        outputs: [
+          {
+            referenceName: DataFactoryDataSetOutName
+            type: 'DatasetReference'
+          }
+        ]
+      })
+    ]
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep:138]: same comment as above...

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep (hunk 2 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep:68]: used .name notation

**Comment by azure-quickstarts** [comment]
@djpmsft - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**